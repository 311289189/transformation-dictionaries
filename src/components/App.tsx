import * as React from 'react';
import './App.css';

export default class App extends React.Component {
    render() {
        return (
            <div className='app'>
                <h1>Transformation Dictionaries</h1>
            </div>
        );
    }
}
