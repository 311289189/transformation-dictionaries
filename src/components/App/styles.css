.app {
  font-family: helvetica, arial, sans-serif;
  padding: 2em;
}

.app > .section {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.resultsArea {
  padding-top: 40px;
}
