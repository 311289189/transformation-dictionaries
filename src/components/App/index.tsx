import * as React from 'react'
import classnames from 'classnames'
import styles from './styles.css'
import DataTable from '../DataTable'

export default () => (
    <main className={styles.app}>
        <section className={styles.section}>
            <h1>Transformation Dictionaries</h1>
        </section>
        <section className={styles.section}>
            <h3>Dictionary</h3>
            <DataTable editable />
        </section>
        <section className={classnames(styles.resultsArea, styles.section)}>
            <h3>Results</h3>
            <DataTable />
        </section>
    </main>
)
