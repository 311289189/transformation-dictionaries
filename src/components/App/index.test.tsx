import * as React from 'react'
import Store from '../../store'
import App from './index'
import { Provider } from 'mobx-react'
import { render } from 'enzyme'

it('App', () => {
    describe('matches snapshot', () => {
        const wrapper = render(
            <Provider store={Store}>
                <App />
            </Provider>
        )

        expect(wrapper).toMatchSnapshot()
    })
})
