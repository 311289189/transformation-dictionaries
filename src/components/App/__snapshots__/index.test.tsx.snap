// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
<main
  class="app"
>
  <section
    class="section"
  >
    <h1>
      Transformation Dictionaries
    </h1>
  </section>
  <section
    class="section"
  >
    <h3>
      Dictionary
    </h3>
    <table
      class="dataTable"
    >
      <tbody>
        <tr
          class="row editable"
        >
          <td>
            <form>
              <input
                class="inputCell"
                required=""
                type="text"
                value=""
              />
            </form>
          </td>
          <td>
            <form>
              <input
                class="inputCell"
                required=""
                type="text"
                value=""
              />
            </form>
          </td>
          <td
            class="controls editable"
          >
            <img
              alt="Submit"
              class="controlButton"
              src="[object Object]"
            />
            <img
              alt="Delete"
              class="controlButton"
              src="[object Object]"
            />
          </td>
        </tr>
      </tbody>
    </table>
  </section>
  <section
    class="resultsArea section"
  >
    <h3>
      Results
    </h3>
    <table
      class="dataTable"
    >
      <tbody>
        <tr
          class="explainerText"
        >
          <td>
            <div>
              Nothing to display.
            </div>
            <br />
            <div>
              Add dictionary items using the cells above, all cells are WSYWYG and can be updated individually, use the green button or hit Enter to submit an entry. Use the red button to delete rows.
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </section>
</main>
`;
